Make patch set root to drive on windows

jrd-rocks opened the pull request:
Set the drive letter on windows so patching works when temp file and scala file are on different drives. It should revert to roor on linux. _NOT tested on linux, it might kill the extension there._

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ensime/ensime-sublime?pullRequest=132) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ensime/ensime-sublime?pullRequest=132) before we can accept your contribution.<br/><hr/>**Jeroen** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ensime/ensime-sublime?pullRequest=132) it.</sub>

VC1995 commented on ensimesublime/protocol.py:
```diff
@@ -238,7 +239,13 @@ def apply_refactor(self, call_id, payload):
+        root = path.drive or path.root
+        self.env.logger.debug("Refactoring set root: {}"
+                                 .format(root))
+        result = patch_set.apply(0, root)
```
You might want to pass str(root) to avoid `TypeError: argument should be string, bytes or integer, not PosixPath`

VC1995 commented on ensimesublime/protocol.py:
```diff
@@ -238,7 +239,13 @@ def apply_refactor(self, call_id, payload):
                                     .format(diff_file))
             return
-        result = patch_set.apply(0, "/")
+        path =  Path(self.refactorings[payload['procedureId']])
```
Using `PurePath` would be more suitable as it can't access the os and we don't want it to anyway. What do you think?

jrd-rocks commented:
You're right, I tried do make things cleaner. I removed pathlib entirely and just added the needed code to the already existing paths.py. For non windows environments bypassing the logic completely. I added a (very) few tests that test that root or drive is returned when needed.

VC1995 commented:
:+1:
